api/nodes/views.py

```diff
@@ -1212,7 +1222,16 @@ def get_queryset(self):
         if not hasattr(node_addon, 'get_folders'):
             raise EndpointNotImplementedError('Endpoint not yet implemented for this addon')
 
-        return node_addon.get_folders(path=path, folder_id=folder_id)
+        #  Convert v1 errors to v2 as much as possible.
+        try:
+            return node_addon.get_folders(path=path, folder_id=folder_id)
+        except InvalidAuthError:
+            raise NotAuthenticated('This add-on could not be authenticated.')
+        except HTTPError as exc:
+            v2_exception = HTTP_CODE_MAP.get(exc.code)
+            if v2_exception:
+                raise v2_exception
+            raise UnknownError()
```

Comment:
I'm not sure we should re-raise as a DRF exception in the case of an unexpected error. If the error is unexpected, we should fail loudly so that we can see all possible info in Sentry. If we swallow the error like this, it will be harder to debug. Something like

```python
exception = HTTP_CODE_MAP.get(exc.code, exc)
raise exception
```